### PR TITLE
Fix for EthCall reverts (#902)

### PR DESCRIPTION
**Pull request by kostind**
### What does this PR do?
Fix EthCall reverts and getRevertReason methods.

### Where should the reviewer start?
EthCall
EthCallIT

### Why is it needed?
If you need to execute your contract methods which update the state without transaction and get the reason for the transaction failure.

### CODE REVIEW

**Comment by kostind** [comment]
@iikirilov Should I rebase my fork and recreate pull request in order to avoid commits like " Merge pull request #1 from web3j/master "?

**Comment by iikirilov** [comment]
> @iikirilov Should I rebase my fork and recreate pull request in order to avoid commits like " Merge pull request #1 from web3j/master "?

yes

File: core/src/main/java/org/web3j/protocol/core/methods/response/EthCall.java [hunk 1]
```diff
@@ -38,15 +38,21 @@ public String getValue() {
     }
 
     public boolean reverts() {
```
Comment (iikirilov) [inline core/src/main/java/org/web3j/protocol/core/methods/response/EthCall.java]: I prefer this method to be renamed to `isReverted`

**Comment by kostind** [comment]
> LGTM - one suggestion for method name change.

Done, thank you.

**Comment by iikirilov** [comment]
Yes, I prefer a rebase and force-push, but either way is fine as I will squash and merge.

I think this will clash with #1026. Is it possible to extend this for ethSendTransaction also?

**Comment by kostind** [comment]
> Yes, I prefer a rebase and force-push, but either way is fine as I will squash and merge.
> 
> I think this will clash with #1026. Is it possible to extend this for ethSendTransaction also? Maybe I just need to merge both PRs?

I will do rebase before creating PR next time. Thanks for the clarification.

Looks like it will clash only with method `EthCall.reverts` which has been renamed to `isReverted`.
I think we have a few options.

1) Rename method back to `reverts`. Probably, it's a good option because all projects which depend on web3j's EthCall have to use `isReverted` instead of `reverts` after web3j update.
2) Add method `reverts` which simple use `isReverted` for backward compatibility.
3) Merge both PRs as is (#1026 first), fix errors (TransactionManager.assertCallNotReverted, ReadonlyTransactionManagerTest.sendCallRevertedTest)

I can help with either variant.

**Comment by iikirilov** [comment]
ok cool. let's go with option 4. don't change anything in the PR :)

**Comment by kostind** [comment]
Done.
I'm not sure how to fix the formatting issue.

**Comment by iikirilov** [comment]
@kostind formatting fix with `./gradlew spotlessApply`

**Comment by kostind** [comment]
@iikirilov Thanks.

**Pull request merged**